common/aws/cli.go

```diff
@@ -3,20 +3,48 @@ package aws
 import (
 	"github.com/Layr-Labs/eigenda/common"
 	"github.com/urfave/cli"
+	"time"
 )
 
 var (
-	RegionFlagName          = "aws.region"
-	AccessKeyIdFlagName     = "aws.access-key-id"
-	SecretAccessKeyFlagName = "aws.secret-access-key"
-	EndpointURLFlagName     = "aws.endpoint-url"
+	RegionFlagName                      = "aws.region"
+	AccessKeyIdFlagName                 = "aws.access-key-id"
+	SecretAccessKeyFlagName             = "aws.secret-access-key"
+	EndpointURLFlagName                 = "aws.endpoint-url"
+	FragmentPrefixCharsFlagName         = "aws.fragment-prefix-chars"
+	FragmentParallelismFactorFlagName   = "aws.fragment-parallelism-factor"
+	FragmentParallelismConstantFlagName = "aws.fragment-parallelism-constant"
+	FragmentReadTimeoutFlagName         = "aws.fragment-read-timeout"
+	FragmentWriteTimeoutFlagName        = "aws.fragment-write-timeout"
 )
 
 type ClientConfig struct {
-	Region          string
-	AccessKey       string
+	// Region is the region to use when interacting with S3. Default is "us-east-2".
+	Region string
+	// AccessKey to use when interacting with S3.
+	AccessKey string
+	// SecretAccessKey to use when interacting with S3.
 	SecretAccessKey string
-	EndpointURL     string
+	// EndpointURL of the S3 endpoint to use. If this is not set then the default AWS S3 endpoint will be used.
+	EndpointURL string
+
+	// FragmentPrefixChars is the number of characters of the key to use as the prefix for fragmented files.
+	// A value of "3" for the key "ABCDEFG" will result in the prefix "ABC". Default is 3.
+	FragmentPrefixChars int
+	// FragmentParallelismFactor helps determine the size of the pool of workers to help upload/download files.
+	// A non-zero value for this parameter adds a number of workers equal to the number of cores times this value.
+	// Default is 8. In general, the number of workers here can be a lot larger than the number of cores because the
+	// workers will be blocked on I/O most of the time.
+	FragmentParallelismFactor int
+	// FragmentParallelismConstant helps determine the size of the pool of workers to help upload/download files.
+	// A non-zero value for this parameter adds a constant number of workers. Default is 0.
+	FragmentParallelismConstant int
+	// FragmentReadTimeout is used to bound the maximum time to wait for a single fragmented read.
+	// Default is 30 seconds.
+	FragmentReadTimeout time.Duration
+	// FragmentWriteTimeout is used to bound the maximum time to wait for a single fragmented write.
+	// Default is 30 seconds.
+	FragmentWriteTimeout time.Duration
```

Comment:
Your suggestion makes sense. The core difficulty is that `aws.ClientConfig` is currently used to build an S3 client by `disperser/cmd/apiserver`, `disperser/cmd/batcher`, and `disperser/cmd/dataapi`. Even harder is the fact that these use cases use the `aws.ClientConfig` object to instantiate both an s3 client and a dynamoDB client, meaning that if I nest `aws.ClientConfig` inside S3 config then it gets duplicated.

I'm sure it's possible to untangle the mess I describe above, but my concern is scope creep for this PR. I've got some ideas for how we might simplify the way we manage our project's configuration, would you be interested in discussing this offline?

Reply:
Yes, interested in discussing, no need for scope creep I agree, since these are only used by us internally so its fine to have breaking changes.

That being said, I don't see why having the duplicate ClientConfig is bad. In fact, one could want an s3 client and dynamoDB client with different aws configs (say the dbs live in different regions for eg).